src/mono/browser/runtime/loader/assets.ts

```diff
@@ -250,7 +285,11 @@ export function prepareAssets() {
             mono_assert(!asset.resolvedUrl || typeof asset.resolvedUrl === "string", "asset resolvedUrl could be string");
             mono_assert(!asset.hash || typeof asset.hash === "string", "asset resolvedUrl could be string");
             mono_assert(!asset.pendingDownload || typeof asset.pendingDownload === "object", "asset pendingDownload could be object");
-            assetsToLoad.push(asset);
+            if (isCoreAssembly(asset)) {
```

Comment:
perhaps here we could assume that if somebody is using `config.assets` everything is "core" and drop `isCoreAssembly`

Reply:
Exactly